Improve release document (#415)
kyma-project/community: guidelines/releases-guidelines/02-kyma-release-process.md

   > **CAUTION:** Never use `/test all` as it might run tests that you do not want to execute.

> 3. Execute remaining tests. There are dependencies between jobs, so the order described below is important.

Reviewer: ```suggestion
3. Execute remaining tests. There are dependencies between jobs, so follow the provided order of steps.
```
